app/javascript/components/public_health/query/AdvancedFilter.js

```diff
@@ -263,19 +264,41 @@ class AdvancedFilter extends React.Component {
   apply = () => {
     this.setState({ show: false, applied: true }, () => {
       this.props.advancedFilterUpdate(this.state.activeFilterOptions);
+      if (this.props.updateStickySettings && this.state.activeFilter) {
+        localStorage.setItem(`SaraFilter`, this.state.activeFilter.id);
+      }
     });
   };
 
   // Clear the current filter
   clear = () => {
-    this.setState({ activeFilter: null, applied: false }, () => {
+    this.setState({ activeFilterOptions: [], show: false, activeFilter: null, applied: false }, () => {
+      this.add();
       this.props.advancedFilterUpdate(this.state.activeFilter);
       if (this.props.updateStickySettings) {
         localStorage.setItem(`SaraFilter`, null);
       }
     });
   };
 
+  // Reset modal when cancelled
+  cancel = () => {
+    // check if there was a filter applied
+    const appliedFilterId = localStorage.getItem(`SaraFilter`);
+    const appliedFilter = this.state.savedFilters.find(filter => filter.id === parseInt(appliedFilterId));
```

Comment:
This is not sufficient to resolve this issue:
>Currently if you have a filter applied and then select a saved advanced filter (i.e., from the drop-down) and click "Cancel": The modal closes and you return to the dashboard (the filter you created is not applied)–this is correct behavior. When you then click "Advanced Filter" it shows the parameters you had previously entered–this is incorrect behavior. **The modal should revert to its previous state (the last filter that was applied, or blank if no filter was applied)**

This only checks if there was a saved filter applied, but if there was an unsaved filter applied I would expect that to show up in the above case as well. To replicate what I mean, do the following:

1. Apply an unsaved filter.
2. Select a saved filter -> click cancel
3. Clicked "Advanced Filter" and notice it's now blank even though there is an unsaved filter applied.

Reply:
If we want to be able to do that we are going to need to store a previous state of some kind that holds the last applied state... will work on that

Reply:
this should be fixed now - it's a better solution too which is great